lua/lspconfig.lua

```diff
@@ -72,9 +70,12 @@ end
 local mt = {}
 function mt:__index(k)
   if configs[k] == nil then
-    -- dofile is used here as a performance hack to increase the speed of calls to setup({})
-    -- dofile does not cache module lookups, and requires the absolute path to the target file
-    pcall(dofile, script_path .. 'lspconfig/' .. k .. '.lua')
+    local success, config = pcall(require, 'lspconfig/server_configurations/' .. k)
+    if success then
+      configs[k] = config
+    else
+      vim.notify(string.format("Configuration not available for %s", k))
```

Comment:
```suggestion
      vim.notify(string.format("Configuration not available for %s", k), vim.log.levels.WARN)
```
Although I'm not really sure what the point here is.

Reply:
The point should be obvious by the error message.

Reply:
Maybe the error message should suggest that the provided server name doesn't exist? I'm thinking some users might interpret the current error message as something that can be fixed, as the definition of "available" is a bit open for interpretation. Idk maybe something along the lines of `Unable to retrieve configuration for server %s because the server doesn't exist.`

Reply:
And this message might get noisy for 3rd party code that checks for the existence of servers (I think this is somewhat common for custom servers):

```lua
if not configs.custom_server then
  configs.custom_server = {...}
end
```

Reply:
> 

Shouldn't apply in this case. This metamethod will only affect indexing the lspconfig table, not the configs table.